jooby/src/main/java/io/jooby/Context.java

```diff
@@ -627,6 +628,13 @@
    */
   @Nonnull String getProtocol();
 
+  /**
+   * The certificates presented by the client for mutual TLS. Empty if ssl is not enabled, or client authentication is not required.
+   *
+   * @return The certificates presented by the client for mutual TLS. Empty if ssl is not enabled, or client authentication is not required.
+   */
+  @Nonnull Certificate[] getClientCertificates();
```

Comment:
I do prefer `List<Certificate>` over `Certificate[]`

Reply:
The underlying method that gets called is getPeerCertificates() and it returns an array. (https://docs.oracle.com/en/java/javase/11/docs/api/java.base/javax/net/ssl/SSLSession.html#getPeerCertificates() for netty
https://undertow.io/javadoc/2.1.x/io/undertow/server/SSLSessionInfo.html#getPeerCertificates--  for undertow
For jetty, you already add the `SecureRequestCustomizer())`, which adds that javax.servlet.request.X509Certificate as the key, and puts the value in the map as an array)
The order of the certificates in the array matters. From the java documentation (first link) 
```
Returns:
an ordered array of peer certificates, with the peer's own certificate first followed by any certificate authorities.
```
If you return a List, it could not be a generic List. It would have to be an implementation of List that maintains order. Making this change would introduce the risk of messing up the order during the transfer from array to the list, and most callers would expect it to be an array.  Your decision. 


Side note, I do see from https://www.eclipse.org/jetty/javadoc/jetty-11/constant-values.html#org.eclipse.jetty.server.SecureRequestCustomizer.JAKARTA_SERVLET_REQUEST_X_509_CERTIFICATE that as of Jetty 11, it should be `jakarta.servlet.request.X509Certificate`. I can just use the static field.  I can make this update when you decide whether this return type should remain an array, or change to a list.

Reply:
List in general are better than array. A call to: `Arrays.asList()` will keep the order.

No jetty 11 for now. I'm planning to make 3.x to require at least Java 11 and definitely upgrade jetty to 11 or whatever.